plugins/modules/cloud/google/gcp_storage_aws_transfer_job.py

```diff
@@ -0,0 +1,272 @@
+    start_time = datetime.datetime.strptime(module.params['scheduled_start_time_utc'], '%H:%M')
+
+    delete_transfer_job = {
+        'project_id': module.params['project_id'],
+        'transfer_job': {
+            'status': 'DELETED'
+        }
+    }
+
+    transfer_job = {
+        'description': module.params['description'],
+        'status': 'ENABLED',
+        'projectId': module.params['project_id'],
+        'schedule': {
+            'scheduleStartDate': {
+                'day': start_date.day,
+                'month': start_date.month,
+                'year': start_date.year
+            },
+            'scheduleEndDate': {
+                'day': start_date.day,
+                'month': end_date.month,
+                'year': end_date.year
+            },
+            'startTimeOfDay': {
+                'hours': start_time.hour,
+                'minutes': start_time.minute,
+                'seconds': start_time.second
+            }
+        },
+        'transferSpec': {
+            'objectConditions': {
+                'includePrefixes': module.params['aws_s3_bucket_prefix']
+            },
+            'transferOptions': {
+                'overwriteObjectsAlreadyExistingInSink': 'False',
+                'deleteObjectsUniqueInSink': 'False',
+                'deleteObjectsFromSourceAfterTransfer': 'False'
+            },
+            'awsS3DataSource': {
+                'bucketName': module.params['aws_s3_bucket'],
+                'awsAccessKey': {
+                    'accessKeyId': module.params['aws_access_key'],
+                    'secretAccessKey': module.params['aws_secret_key']
+                }
+            },
+            'gcsDataSink': {
+                'bucketName': module.params['gcp_storage_bucket']
+            }
+        }
+    }
+
+    credentials = service_account.Credentials.from_service_account_file(module.params['service_account_file'])
+    storagetransfer = googleapiclient.discovery.build('storagetransfer', 'v1', credentials=credentials)
+
+    filterString = (
+        '{{"project_id": "{project_id}"}}'
+    ).format(project_id=module.params['project_id'])
+
+    queryResult = storagetransfer.transferJobs().list(filter=filterString).execute()
+    result['message'] = queryResult
+
+    # Delete existing job if exists, then create new job
+    if 'transferJobs' in queryResult:
+        for transferJob in queryResult['transferJobs']:
+            if transferJob['description'] == module.params['description']:
+                storagetransfer.transferJobs().patch(body=delete_transfer_job, jobName=transferJob['name']).execute()
+                result['message'] = 'Deleted existing Transfer Jobs with the same Description'
+
+    if 'present' in module.params['state']:
+        storagetransfer.transferJobs().create(body=transfer_job).execute()
+        result['message'] = 'Created new Transfer Job'
+
+    result['original_message'] = module.params
```

Comment:
I would not return the module parameters. This can accidentally leak credentials (passed in the module parameters).

Reply:
(Also, you documented that return value differently. The return docs are talking about a *name*, not about all parameters.)